lxc/profile.go

```diff
@@ -278,14 +281,28 @@ func (c *cmdProfileCopy) Run(cmd *cobra.Command, args []string) error {
 		dest.name = source.name
 	}
 
-	// Copy the profile
-	profile, _, err := source.server.GetProfile(source.name)
+	sourceProfile, _, err := source.server.GetProfile(source.name)
 	if err != nil {
 		return err
 	}
 
+	// update if refresh flag is set and the destination profile exist
```

Comment:
You can make this simpler without having to call `dest.server.GetProfile()` twice:

```go
if dest.name == "" {
	dest.name = source.name
}

// Copy the profile
profile, _, err := source.server.GetProfile(source.name)
if err != nil {
	return err
}

// Refresh the profile if requested.
if c.flagRefresh {
	return dest.server.UpdateProfile(dest.name, profile.Writable(), "")
}

newProfile := api.ProfilesPost{
	ProfilePut: profile.Writable(),
	Name:       dest.name,
}
```

Reply:
It's a good idea to not call GetProfile twice but the suggested change does not work. If you call try `copy --refresh` and the destination profile does not exist, you'll get an error because `dest.server.UpdateProfile(dest.name, profile.Writable(), "")` expect the destination profile to exist. So we have to check if the destination profile exist. I was thinking of using `dest.server.GetProfileNames()` and check in the returned list. Do you have any other suggestion ?

Reply:
Yes, you are right. There is one other `--refresh` option for the `lxc copy` command which also continues if the target instance isn't there yet. So I think we should keep this behavior then. 

Regarding the change, just append the check for `http.StatusNotFound`, return the error if it's not 404, otherwise continue on with the profile creation:

```go
if c.flagRefresh {
	err := dest.server.UpdateProfile(dest.name, profile.Writable(), "")
	if !api.StatusErrorCheck(err, http.StatusNotFound) {
		return err
	}
}
```

This way you assume the profile already exists, if not create it.